main/src/library/MutSet.flix

```diff
@@ -338,22 +332,22 @@ namespace MutSet {
     ///
     /// The function `f` must be pure.
     ///
-    pub def partition(f: a -> Bool, s: MutSet[a]): (MutSet[a], MutSet[a]) & Impure with Order[a] =
+    pub def partition(f: a -> Bool, s: MutSet[a, r]): (MutSet[a, r], MutSet[a, r]) & reg(r) with Order[a] =
         let MutSet(ms) = s;
         let (ys, zs) = Set.partition(f, deref ms);
-            (MutSet(ref ys), MutSet(ref zs))
+            (MutSet(ref ys), MutSet(ref zs)) as (MutSet[a, r], MutSet[a, r]) & reg(r)
```

Comment:
I think I have an idea for this. Thanks for discovering it :)

Reply:
If you rebase onto master then you can add:

```
instance Scoped[MutSet2[t]] {
    pub def regionOf(_: MutSet2[t, r]): Region[r] = () as Region[r]
}
```

and then you can write e.g.:

```
    pub def copy(s: MutSet2[a, r]): MutSet2[a, r] & reg(r) = 
        let MutSet2(ms) = s;
        MutSet2(ref (deref ms) @ Scoped.regionOf(s))
```

This is a way to get a region from an argument.
```